torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -582,6 +582,17 @@ std::shared_ptr<SugaredValue> toSugaredValue(
   } else if (
       obj.ptr() == py::module::import("torch.jit").attr("annotate").ptr()) {
     return SpecialFormValue::create(prim::annotate);
+#ifdef USE_DISTRIBUTED
+  } else if (
+      // RPC module is only avaialble for Python3
+      // when build flag "USE_DISTRIBUTED" is on.
+      !py::module::import("torch._six").attr("PY2").cast<bool>() &&
+      obj.ptr() ==
+          py::module::import("torch.distributed.rpc")
+              .attr("rpc_async")
+              .ptr()) {
+    return SpecialFormValue::create(prim::rpc_async);
```

Comment:
What happens to the existing `rpc_async` in rpc/api.py?

Reply:
@pritamdamania87 

This is telling the JIT front end that, for this python call, it want's to be special handled, so it's dispatched to JIT operator, prim::rpc_async, so the exiting Python rpc_async entry is left aside for TorchScript.

We can make Python code path and TorchScript code path behave exactly the same by moving the peripheral logic to the RPC C++ entry.

I have already moved `args = args if args else ()` and `kwargs = kwargs if kwargs else ()`.

Reply:
> We can make Python code path and TorchScript code path behave exactly the same by moving the peripheral logic to the RPC C++ entry.

If we do that I'm guessing `rpc_async` can't support calling arbitrary python functions right? If it still can, then let's merge the codepaths. It is confusing to have two separate implementations of rpc_async in the same package that behave differently.